### DISCUSSION TITLE
Texture for ImGui

### DISCUSSION THREAD

**Discussion by wecand0** [maplibre/maplibre-native#1530] [Q&A] [answered]
I want to get texture from framebuffer and to render this **texture** like this https://github.com/ocornut/imgui/wiki/Image-Loading-and-Displaying-Examples#example-for-opengl-users
Is it possible to get it? The platform: linux+glfw

As I can understand in platform/glfw/glfw_gl_backend.cpp there is GLFWGLRenderableResource, in this class we can get context where there is method getContext(), am I in the right way?

**Comment by louwers** [comment]
Hi @wecand0 did you manage to figure it out?

**Reply by wecand0** [reply]
still not, I reckon that I need this ID from here https://github.com/maplibre/maplibre-native/blob/9cfb69539bbf9c8d64c7a12f6e22a2133dba5426/src/mbgl/gl/texture2d.cpp#L116
It it right approach?
PS I know we had conversation before, I wrote additional SDK for developing for Linux+glfw(implemented drawing any geojson(circles with radius, hexagons and etc), if it is fruitful for project - a bit later I can share my source code.

May be it would be very great to integrate ImGui in maplibre-native as it is made for thousands projects for debugging+custom interface.
I found this project with ImGui + maps -> https://github.com/tangrams/tangram-es/tree/main/platforms/common but it is a bit rip

**Reply by louwers** [reply, marked as answer]
You can use a CustomLayer if you want to have low level access into the rendering loop of MapLibre Native and draw stuff with OpenGL yourself. Here is an example:

https://github.com/maplibre/maplibre-native/blob/10955fbd41554760682c593c759cdace5a7858a9/test/api/custom_layer.test.cpp#L39
